src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,201 @@
+import React, {
+  CSSProperties,
+  FC,
+  HTMLAttributes,
+  useEffect,
+  useState
+} from 'react';
+import { classNames } from '../../utils/classNames';
+import Icon from '../Icon';
+import { IconSize } from '../Icon/Icon';
+import {
+  convertSizeNameToSizeNumber,
+  getCurrentHalfStars,
+  getCurrentStarRating,
+  getDefaultHalfStars,
+  getFullStars,
+  getIconColor,
+  getIconName,
+  getSizeName
+} from './helper';
+import { useCurrentRatingLabel } from './hooks';
+
+export interface IRatingProps {
+  /** set className to the component */
+  className?: string;
+  /** set rating to be disabled */
+  disabled?: boolean;
+  /** set rating to be readonly */
+  readonly?: boolean;
+  /** set number of stars */
+  ratingCount?: number;
+  /** set size of stars */
+  size?: IconSize;
+  /** controll value of filled star ratings (rating value can be .5 fraction when "half" prop is added) */
+  defaultRating?: number;
+  /** allow half fraction of stars */
+  half?: boolean;
+  /** set label with satisfaction words after the rating component */
+  isLabel?: boolean;
+  /** set inline style to the component */
+  style?: CSSProperties;
+  /** get the current rating value outside of component */
+  onChange?: (rating: any) => void;
+}
+
+export interface IStar {
+  value: number;
+  index: number;
+}
+
+export type IStars = IStar[];
+
+export type RatingProps = IRatingProps & HTMLAttributes<HTMLElement>;
+
+/**
+ * Ratings provide insight regarding others' opinions and experiences, and can allow the user to submit a rating of their own.
+ *
+ * ```js
+ * import {Rating} from 'pat-ui'
+ * ```
+ */
+
+export const Rating: FC<RatingProps> = (props) => {
+  const {
+    className,
+    disabled,
+    readonly,
+    ratingCount,
+    size,
+    defaultRating,
+    isLabel,
+    half,
+    style,
+    onChange,
+  } = props;
+
+  // managing classnames
+  let styleClasses = classNames('rating_icon-group', {
+    disabled: !!disabled,
+    readonly: !!readonly,
+  });
+  if (className) {
+    styleClasses += ' ' + className;
+  }
+  // rating array as the default to stars state
+  const starArr = new Array(ratingCount).fill(0).map((item, index) => {
+    return {
+      value: 0,
+      index: index + 1,
+    };
+  });
+
+  // states to manage save current rating value,currentHover position, and stars value
+  const [currentTotalRating, setCurrentTotalRating] = useState(defaultRating);
+  const [currentHovering, setCurrentHovering] = useState<number>(-1);
+  const [stars, setStars] = useState<IStars>(starArr);
+
+  // custom hook to track the current rating label
+  const { labelName } = useCurrentRatingLabel(
+    currentTotalRating as number,
+    ratingCount as number
+  );
+
+  const handleOnClick = () => {
```

Comment:
Because there is only single onClick and onMouse/Leave events in my component, I may leave them this way for now. Once I found good names I will definitely consider to replace them.